### ISSUE TITLE
Create hashcode and equals methods for jaxb generated classes

### ISSUE THREAD

**Issue by jkranes** [smart-cow/scow#9]
These would be helpful for the agileclient code.

**Comment by brosenberg42** [comment]
I found the necessary maven plugin and configuration to generate hashCode and equals methods (as well as toString, copyable, mergeable). The plugin's documentation is at http://confluence.highsource.org/display/J2B/JAXB2+Basics+Plugins. Since we haven't yet decided which fields will be used to determine equality, I made a separate branch with the [code that can be used to generate the equals and hashCode methods](https://github.com/smart-cow/scow/blob/a318a/cow-server-api/pom.xml#L32-110).

We now need to decide which fields are included in determining equality. The problem is basically if a person puts on a hat, one of its fields has changed, but it is still equal. In general the fields used to determine equality should be immutable. Those fields should be set when the object is constructed, and it should not be possible to change them over the lifetime of the object.

Getting this right is important because many of the Java built-in collections use the equals and hashCode methods on the elements of the collection. If they are not implemented correctly, collections of the jaxb classes may not operate correctly.
